packages/app/src/cli/services/context.ts

```diff
@@ -147,6 +147,169 @@ export async function ensureGenerateContext(options: GenerateContextOptions): Pr
   }
 }
 
+interface ReusedLogsValuesOptions {
+  organization: Organization
+  selectedApp: OrganizationApp
+  cachedInfo?: CachedAppInfo
+}
+
+export interface LogsContextOptions {
+  directory: string
+  apiKey?: string
+  storeFqdns?: string[]
+  reset: boolean
+  developerPlatformClient: DeveloperPlatformClient
+}
+
+interface LogsContextOutput {
+  remoteApp: Omit<OrganizationApp, 'apiSecretKeys'> & {apiSecret?: string}
+  storeIds: string[]
+  localApp: AppInterface
+}
+
+/**
+ * Make sure there is a valid context to execute `logs`
+ * That means we have a valid organization and app.
+ *
+ * If there is an app from flags, we check if it's valid. If it's not, throw an error.
+ * If there is info in the cache or current configuration, check if it is still valid and return it.
+ * If there is no info (or is invalid):
+ *  - Show prompts to select an org and app
+ *  - The info will be updated in the cache or current configuration
+ *
+ * @param options - Current dev context options
+ * @returns The selected org, app and dev store
+ */
+export async function ensureLogsContext(options: LogsContextOptions): Promise<LogsContextOutput> {
```

Comment:
There's a lot of duplicated logic between this function and `ensureDevContext`. I want to spend some time looking into some refactors to keep the two implementations more synchronized over time while still allowing logs to diverge in how it accepts shops